Referencing event toggling to performance page

AlexJerabek opened the pull request:
The preview section on event toggling seems like a better fit for the advanced concept page on performance, rather than the basic page about events.

Also reworded the text to better match the new context.

o365devx commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-events.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-EventPerf/docs/excel/excel-add-ins-events.md) | :white_check_mark:Succeeded | |
[docs/excel/performance.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-EventPerf/docs/excel/performance.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C8%5C30%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201808302103039395-349%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=aAAtaOSpMfOcP1FvYuZo6LO8fH9zDii8XolhMP1grIM%3D&st=2018-08-30T20%3A59%3A14Z&se=2018-09-30T21%3A04%3A14Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

AlexJerabek commented:
@kbrandl My rationale here is the only reasons to turn off events are performance related. Maybe I'm missing something obvious, but I don't think there's a scenario in which I'd want events off as part of my app flow.

o365devx commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-events.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-EventPerf/docs/excel/excel-add-ins-events.md) | :white_check_mark:Succeeded | |
[docs/excel/performance.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-EventPerf/docs/excel/performance.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C8%5C31%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201808310001138991-349%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=DhP6%2BbGMkt0WgGJt0P0xxdHJc%2FiJBeKc3KFTH3TED0w%3D&st=2018-08-30T23%3A57%3A29Z&se=2018-10-01T00%3A02%3A29Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

o365devx commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-events.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-EventPerf/docs/excel/excel-add-ins-events.md) | :white_check_mark:Succeeded | |
[docs/excel/performance.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-EventPerf/docs/excel/performance.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C8%5C31%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201808310009467967-349%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=9AaNPXZ%2BiKm2N6eax6Fg0C%2BwJ5Z1PAloGUg8Vo%2BhtgI%3D&st=2018-08-31T00%3A05%3A56Z&se=2018-10-01T00%3A10%3A56Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

AlexJerabek commented:
@kbrandl Thanks for the feedback. I've updated the articles (and the PR title) accordingly.

I also fixed up the links to point to the new ref docs.

o365devx commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-events.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-EventPerf/docs/excel/excel-add-ins-events.md) | :white_check_mark:Succeeded | |
[docs/excel/performance.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-EventPerf/docs/excel/performance.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C8%5C31%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201808310012440064-349%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=GR%2F9LMQ%2FxOWHnYl646HIzVlyNkyHJvKveYqYBTC%2FqzE%3D&st=2018-08-31T00%3A08%3A56Z&se=2018-10-01T00%3A13%3A56Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

kbrandl commented on docs/excel/excel-add-ins-events.md:
```diff
@@ -133,27 +133,27 @@ function remove() {
 > This feature is currently available only in public preview (beta). To use it, you must reference the beta library of the Office.js CDN: https://appsforoffice.microsoft.com/lib/beta/hosted/office.js.
 
-Events are turned on and off at the [runtime](https://docs.microsoft.com/en-us/javascript/api/excel/excel.runtime?view=office-js) level. 
+Some scenarios benefit from the performance boost given by turning off events. 
```
Suggest that you use "disabling events" instead of "turning off events" (to match the lingo used in the section heading).  

Also, you might consider rewording to something more along the lines of: "In some scenarios, performance of an add-in may be improved by disabling events."

kbrandl commented on docs/excel/excel-add-ins-events.md:
```diff
@@ -133,27 +133,27 @@ function remove() {
 
-Events are turned on and off at the [runtime](https://docs.microsoft.com/en-us/javascript/api/excel/excel.runtime?view=office-js) level. 
+Some scenarios benefit from the performance boost given by turning off events. 
+Your app might never need to receive events, or it could ignore events while performing batch-edits of multiple entities. 
```
Maybe start this sentence with "For example, " ?  i.e.,  "For example, your app might..."

kbrandl commented on docs/excel/excel-add-ins-events.md:
```diff
@@ -133,27 +133,27 @@ function remove() {
+Some scenarios benefit from the performance boost given by turning off events. 
+Your app might never need to receive events, or it could ignore events while performing batch-edits of multiple entities. 
+
+Events are turned on and off at the [runtime](https://docs.microsoft.com/javascript/api/excel/excel.runtime) level. 
```
Suggest that you use "enabled and disabled" instead of "turned on and off" (to match the lingo used in the section heading).

kbrandl commented on docs/excel/excel-add-ins-events.md:
````diff
@@ -133,27 +133,27 @@ function remove() {
-        await context.sync();
-    });
-}
+```js
````
The language specified for this code sample is "JavaScript" -- which is what we want, since all other code samples on the page are JavaScript.

However, the code sample itself is TypeScript (e.g., uses `await`).  Suggest modifying the code sample to be pure JavaScript, so that it's consistent with other code samples on this page (and throughout our conceptual doc set).

AlexJerabek replied:
Doesn't JS support await/async? Our out-of-the-box yo office JavaScript file uses it.

```
async function run() {
    try {
            await Excel.run(async context => {
                /**
                 * Insert your Excel code here
                 */
                const range = context.workbook.getSelectedRange();

            // Read the range address
            range.load('address');

            // Update the fill color
            range.format.fill.color = 'yellow';

            await context.sync();
            console.log(`The range address was ${range.address}.`);
            });
        } catch(error) {
            OfficeHelpers.UI.notify(error);
            OfficeHelpers.Utilities.log(error);
        };
}
```

Rick-Kirkham replied:
@AlexJerabek @kbrandl This blog post seems to say that async/await is in ES8 (ES2017) and that it has been released, but I don't necessarily trust blog posts.

kbrandl replied:
ES7 supports `async` / `await` but I believe it requires transpile. Can you clarify this with Michael (and share your learnings with the team via email)?  i.e., what are the requirements for using `async` / `await` in vanilla JavaScript (e.g., does E7 and higher support it, but only with transpile)? This is something we should all understand and be on the same page about. Thanks!

AlexJerabek replied:
I rewrote the sample to avoid using async/await.
We should figure out how to reconcile the new ES7 standard with our docs. It's also something to look into that yo office uses async/await in their JavaScript, but the VS projects do not. There's a need for consistency across those project scaffolds and our docs.

Rick-Kirkham replied:
Why "the new ES7"? ES8 and ES9 have both been released and they support async/await.

kbrandl replied:
Sounds good, thanks @AlexJerabek. +1 on needing to identify the best path forward re using async/await in our docs, and ensuring consistency across both docs and tooling.

AlexJerabek replied:
@Rick-Kirkham Sorry, I meant ES8.

kbrandl commented on docs/excel/performance.md:
```diff
@@ -155,11 +155,18 @@ Excel.run(async (ctx) => {
+
+## Enable and disable events
+
+> [!NOTE]
```
You probably don't need this note here -- since it appears in the Events article where the feature is actually described.

kbrandl commented on docs/excel/performance.md:
```diff
@@ -155,11 +155,18 @@ Excel.run(async (ctx) => {
+> [!NOTE]
+> This feature is currently available only in public preview (beta). To use it, you must reference the beta library of the Office.js CDN: https://appsforoffice.microsoft.com/lib/beta/hosted/office.js.
+
+You can disable events to improve performance. A code sample demonstrating this toggle is in the [Work with Events](excel-add-ins-events.md#enable-and-disable-events) article.
```
Do we know for sure that disabling events ***always*** improves performance?  If not, we might want to phrase this more along the lines of "In some scenarios, performance of an add-in may be improved by disabling events."

Also, suggest replacing "demonstrating this toggle" with "that shows how to enable and disable events".

o365devx commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-events.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-EventPerf/docs/excel/excel-add-ins-events.md) | :white_check_mark:Succeeded | |
[docs/excel/performance.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-EventPerf/docs/excel/performance.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C8%5C31%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201808311816344358-349%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=RI%2FwoIgpdvsoZKue2oYFuDpw%2BBRbNHZco%2FsXDwVjVF0%3D&st=2018-08-31T18%3A12%3A46Z&se=2018-10-01T18%3A17%3A46Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

o365devx commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/excel/excel-add-ins-events.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-EventPerf/docs/excel/excel-add-ins-events.md) | :white_check_mark:Succeeded | |
[docs/excel/performance.md](https://github.com/OfficeDev/office-js-docs-pr/blob/AlexJ-EventPerf/docs/excel/performance.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C8%5C31%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201808311911213839-349%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=7u%2B16WVi5kHikfkNSkqOOAmxiJxfuqfgg7jY8KHP8QA%3D&st=2018-08-31T19%3A07%3A42Z&se=2018-10-01T19%3A12%3A42Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.
